### DISCUSSION TITLE
HyperHDR and WARLS not working?

### DISCUSSION THREAD

**Discussion by DMartin90** [awawa-dev/HyperHDR#870] [General]
Hi All,

I'm trying to migrate from Hyperion to HyperHDR, as I've found the image processing to look FAR better in HyperHDR than Hyperion (which i believe was the whole point of hyperHDR in the first place) The problem I'm having is my LED strip is quite large, I'm using a SK6812 (RGBWW) strip with 764 LEDs (77" TV), and in hyperion the strip is driven with minimal latency over DDP, but when i try to use WARLS in HyperHDR the first half or so of the strip seems to work, and the back half just seems to jump every second or so to catch up so is not smooth (the colours are better tho, i can see that even though the strip is laggy) I believe the problem is the strip isn't recieving WARLS but UDP raw? when i pull up the WLED web interface while streaming with hyperion it reports back "recieving DDP packets", with HyperHDR its saying UDP, which i assume is UDPraw, should the interface say something along the lines of "sound reactive" or "WARLS" ect? 

I have a fairly complicated home automation set up (Almost everything is running though Control4) and i have a network rack in my basement that hosts the media server, as well as the media players, with everythig distiubuted to the TVs via a HDMI/HDBaseT Matrix, the HDMI grabber (a Elgato Game Capture HD60 S+) is connected to one of the spare HDMI outputs on the matrix, this allows me to use one HyperHDR instance and control (via the Control4 system) what source is streaming to HyperHDR and using the JSON API I can turn on/off the LED outputs. Hyperion and HyperHDR are both hosted on my Unraid Server in dockers (not running at the same time obviously) which is in my rack, so moving HyperHDR to a Raspberry Pi for instance and putting it behind the TV isn't really going to work for me, I also have the WLED controllers tied into Control4 to just function as lights when the TV isn't being used, or the bias lighting isn't wanted, so I'd really like to try and make it work with the WLED controllers if possible. 

Doesn't seem like this is a problem for anyone else, so I'm assuming its a problem with my setup somewhere, can anyone who is using WARLS advise what they see on the WLED web interface so I can get a bit closer to a solution? 
If anyone has any suggestions of what could be wrong and what to try tweaking I'm all ears!

**Comment by awawa-dev** [comment]
Hi
UDPraw has ~490 RGB LEDs limit so I'm sure it's WARLS in DNRGB mode (BTW which has also lower network footprint than DDP). Recently there were some seriuos problems with this protocol in WLED itself for example https://github.com/Aircoookie/WLED/issues/3672

Here is our implementation (later first byte is 4 for multi packet WARLS ->DNRGB):
https://github.com/awawa-dev/HyperHDR/blob/aa2007952bbce996ac2b2afaadfe527c558f93f2/sources/leddevice/dev_net/LedDeviceWled.cpp#L273
and here is WLED documentation:
https://github.com/Aircoookie/WLED/wiki/UDP-Realtime-Control#udp-realtime
